pkg/types/types.go

```diff
@@ -75,54 +75,115 @@ const (
-	Pod string `json:"pod,omitempty" column:"pod,template:pod" columnTags:"kubernetes"`
+const (
+	RuntimeNameDocker     RuntimeName = "docker"
+	RuntimeNameContainerd RuntimeName = "containerd"
+	RuntimeNameCrio       RuntimeName = "cri-o"
+	RuntimeNamePodman     RuntimeName = "podman"
+	RuntimeNameUnknown    RuntimeName = "unknown"
+)
 
-	// Container where the event comes from, or empty for host-level or
-	// pod-level event
-	Container string `json:"container,omitempty" column:"container,template:container" columnTags:"kubernetes,runtime"`
+func String2RuntimeName(name string) RuntimeName {
+	switch name {
+	case string(RuntimeNameDocker):
+		return RuntimeNameDocker
+	case string(RuntimeNameContainerd):
+		return RuntimeNameContainerd
+	case string(RuntimeNameCrio):
+		return RuntimeNameCrio
+	case string(RuntimeNamePodman):
+		return RuntimeNamePodman
+	}
+	return RuntimeNameUnknown
+}
+
+type BasicRuntimeMetadata struct {
+	// RuntimeName is the name of the container runtime. It is useful to distinguish
+	// who is the "owner" of each container in a list of containers collected
+	// from multiple runtimes.
+	RuntimeName RuntimeName `json:"runtimeName,omitempty" column:"runtimeName,minWidth:5,maxWidth:12"`
+
+	// ContainerID is the container ContainerID without the container runtime prefix. For
+	// instance, without the "cri-o://" for CRI-O.
+	ContainerID string `json:"containerId,omitempty" column:"containerId,width:13,maxWidth:64"`
+
+	// ContainerName is the container name. In the case the container runtime
+	// response with multiple containers, ContainerName contains only the first element.
+	ContainerName string `json:"containerName,omitempty" column:"containerName,template:container"`
+}
+
+func (b *BasicRuntimeMetadata) IsEnriched() bool {
+	return b.RuntimeName != RuntimeNameUnknown && b.RuntimeName != "" && b.ContainerID != "" && b.ContainerName != ""
+}
+
+type BasicK8sMetadata struct {
+	Namespace     string `json:"namespace,omitempty" column:"namespace,template:namespace"`
+	PodName       string `json:"podName,omitempty" column:"pod,template:pod"`
+	ContainerName string `json:"containerName,omitempty" column:"container,template:container"`
+}
+
+func (b *BasicK8sMetadata) IsEnriched() bool {
+	return b.Namespace != "" && b.PodName != "" && b.ContainerName != ""
+}
+
+type K8sMetadata struct {
+	Node string `json:"node,omitempty" column:"node,template:node"`
+
+	BasicK8sMetadata `json:",inline"`
 
 	// HostNetwork is true if the container uses the host network namespace
 	HostNetwork bool `json:"hostNetwork,omitempty" column:"hostnetwork,hide"`
 }
 
+type CommonData struct {
+	// Runtime contains the container runtime metadata of the container
+	// that generated the event
+	Runtime BasicRuntimeMetadata `json:"runtime,omitempty" column:"runtime" columnTags:"runtime"`
+
+	// K8s contains the Kubernetes metadata of the object that generated the
+	// event
+	K8s K8sMetadata `json:"k8s,omitempty" columnTags:"kubernetes"`
```

Comment:
I'm going to write my comment down but we can handle this in another PR (I know we want to merge this soon to avoid having conflicts again). 

I think both `Runtime` and `K8s` should be pointers. The reason is that events can be generated outside Kubernetes and even outside containers, in that case it's more practical to have a nil pointer rather than an empty structure around.

Reply:
I added it to the TODO list for future PRs. I will create a dedicated issue for them once this is merged.